lmcache/v1/storage_backend/connector/valkey_adapter.py

```diff
@@ -0,0 +1,141 @@
+# SPDX-License-Identifier: Apache-2.0
+# Standard
+from typing import List, Optional, Tuple
+import urllib.parse
+
+# First Party
+from lmcache.logging import init_logger
+from lmcache.v1.storage_backend.connector import (
+    ConnectorAdapter,
+    ConnectorContext,
+    parse_remote_url,
+)
+from lmcache.v1.storage_backend.connector.base_connector import RemoteConnector
+
+logger = init_logger(__name__)
+
+
+class ValkeyConnectorAdapter(ConnectorAdapter):
+    """Unified adapter for all Valkey connection modes"""
+
+    def __init__(self) -> None:
+        super().__init__("valkey://")
+
+    def can_parse(self, url: str) -> bool:
+        """Check if this adapter can handle the given URL."""
+        return url.startswith("valkey://")
+
+    def _parse_database_from_url(self, url: str) -> Optional[int]:
+        database_id: Optional[int] = None
+        if "/" in url:
+            url, db_part = url.split("/", 1)
```

Comment:
It really complex for the parameters parse work. it is not introduced by this PR.

But you may reference other connector implementation and leverage the `extra_config`, it would be much easy to pass some extra_config belongs to `valkey`, something like `valkey_password`, `valkey_username`, `valkey_database_id`.

@sammshen What about your idea?